pkgs/development/tools/build-managers/corrosion/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, rustPlatform
+}:
+
+stdenv.mkDerivation rec {
+  pname = "corrosion";
+  version = "unstable-2021-02-23";
+
+  src = fetchFromGitHub {
+    owner = "AndrewGaspar";
+    repo = "corrosion";
+    rev = "e6c35c7e55a59c8223577b5abc4d253b4a82898b";
+    sha256 = "0vq6g3ggnqiln0q8gsr8rr5rrdgpfcgfly79jwcygxrviw37m44d";
+  };
+
+  patches = [
+    # https://github.com/AndrewGaspar/corrosion/issues/84
+    ./cmake-install-full-dir.patch
```

Comment:
Who wrote this patch? (I assume you did for now.)

Can this be submitted to upstream? Or is it too much of a hacky solution?

Reply:
Yes, me.

@AndrewGaspar is still thinking about the best solution: https://github.com/AndrewGaspar/corrosion/issues/84#issuecomment-763209215.